File: python/day01.py
with open('./inputs/01.txt') as f:
    instructions = f.readline()

def solve(digits, second_part=False):
    size = len(digits)
    jump = 1 if not second_part else size//2
    return sum(n for i, n in enumerate(digits) if n == digits[i-jump])

nums = list(map(int, instructions))
print(solve(nums))
print(solve(nums, second_part=True))

File: nim/day01.nim
import sequtils

const
  digits = readFile("./inputs/01.txt").mapIt(int(it) - int('0'))
  size = digits.len


func solve(jump: int): int =
  for i, n in digits:
    if n == digits[(i + jump) mod size]:
      result += n

echo solve(1)
echo solve(size div 2)
